Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py

```diff
@@ -1523,120 +2415,441 @@ def test_module(client: Client, integration_params: Dict) -> str:
-                        new_last_run[entity_type]['last_created_events'].append(incident_uid)
+                        continue
+
+                    demisto.debug(f"{entity_type} - New incident from object "
+                                  f"last_timestamp : {incident_last_run.get('last_timestamp')} "
+                                  f"/ ID : {incident_last_run.get('id')}")
+                    entity_incidents.append(incident)
+                    new_last_run[entity_type]['last_timestamp'] = incident_last_run.get('last_timestamp')
+                    new_last_run[entity_type]['id'] = incident_last_run.get('id')
+                    # We add this event in the list, as that's a new event we need to remember for next run
+                    last_created_events.append(incident_uid)
 
             if len(entity_incidents) > 0:
                 demisto.info(f"{entity_type} - {len(entity_incidents)} incident(s) to create")
                 incidents += entity_incidents
             else:
-                demisto.info(f"{entity_type} - No new incidents to create, keeping previous last_run data")
+                demisto.debug(f"{entity_type} - No new incidents to create, keeping previous last_run data")
                 new_last_run[entity_type] = previous_last_run[entity_type]
 
             # Update remaining list
             remaining_fetch_types.remove(entity_type)
             if len(remaining_fetch_types) > 0:
-                max_created_incidents = (sanitize_max_results(integration_params.get('max_fetch')) - len(incidents)) \
-                    // len(remaining_fetch_types)
+                max_created_incidents = (max_fetch - len(incidents)) // len(remaining_fetch_types)  # type: ignore
+
+    if is_test:
+        demisto.debug(f"Setting last run to : {new_last_run}")
+        return previous_last_run, []
 
-    demisto.info(f"{len(incidents)} total incident(s) to create")
+    demisto.info(f"{len(incidents)} total incident(s) to create.")
 
     return new_last_run, incidents
 
 
+def get_modified_remote_data_command(client: Client) -> GetModifiedRemoteDataResponse:
+    """
+    Get modified remote data from the Vectra platform and prepare it for mirroring in XSOAR.
+
+    Args:
+        client (Client): An instance of the VectraClient class.
+
+    Returns:
+        GetModifiedRemoteDataResponse: List of incidents IDs which are modified since the last update.
+    """
+    args = demisto.args()
+    command_args = GetModifiedRemoteDataArgs(args)
+    command_last_run_date = dateparser.parse(
+        command_args.last_update, settings={'TIMEZONE': 'UTC'}).strftime('%Y-%m-%dT%H%M')  # type: ignore
+    modified_entities_ids = []
+
+    demisto.debug(f'Last update date of get-modified-remote-data command is {command_last_run_date}.')
+
+    for entity_type in ENTITY_TYPES_FOR_MIRRORING:
+        entity_next_url = None
+        page = 1
+        page_size = 500
+        while True:
```

Comment:
I do not like while True.
Please consider limiting it.
Is there a limit of total to request? 
Can we limit it to 5 iterations? while True and current_iteration < 5 = pages_limit

Reply:
Hi @thefrieddan1, we have provided one if condition in while loop code, using which we will exit from the loop.

1. As we can see, we have used `page_size` parameter for API call, which is set to the `500`.
2. Also, we are collecting all the modified entity IDs in variable `modified_entities_ids`.
3. When the length of `modified_entities_ids` gets larger than `MAX_MIRRORING_LIMIT / 2` which is `2500`, we will exit from loop.
4. So if we consider the edge case for this condition, the while loop will run for max 5-6 round.